### PR TITLE
Update serialbox2 to the new gridtools storage

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
Only review after #25 is merged.

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch jenkins

File: examples/gridtools/simple/example-01-laplacian.cpp [hunk 1]
```diff
@@ -96,19 +97,20 @@ void write() {
...
+  storage_t lap(storage_info, -1., "lap");
 
   std::default_random_engine gen;
   std::uniform_real_distribution<double> dist(0.0, 1.0);
+  auto phi_view = make_host_view(phi);
   for(int i = 0; i < N; ++i)
     for(int j = 0; j < M; ++j)
-      phi(i, j, 0) = dist(gen);
-  
+      phi_view(i, j, 0) = dist(gen);
```
Comment (stefanmoosbrugger) [inline examples/gridtools/simple/example-01-laplacian.cpp]: you could also use the lambda init ctor `storage_t phi(storage_info, [&](int i, int j, int k) { return dist(gen); }, "phi");`. but no need to change this.

File: examples/gridtools/simple/example-02-stateless.cpp [hunk 2]
```diff
@@ -48,28 +48,31 @@ int main() {
...
-    for(int i = 0; i < N; ++i)
-      for(int j = 0; j < M; ++j)
-        for(int k = 0; k < K; ++k)
-          field_in(i, j, k) = dist(gen);
+    {
+      auto view = make_host_view(field_in);
+      for(int i = 0; i < N; ++i)
+        for(int j = 0; j < M; ++j)
+          for(int k = 0; k < K; ++k)
+            view(i, j, k) = dist(gen);
```
Comment (stefanmoosbrugger) [inline examples/gridtools/simple/example-02-stateless.cpp]: same as before.. you could use the lambda.. only if you want.. :)

File: examples/gridtools/simple/example-03-slice.cpp [hunk 3]
```diff
@@ -58,17 +58,19 @@ int main() {
...
-      for(int j = 0; j < M; ++j)
-        for(int k = 0; k < K; ++k)
-          field_in(i, j, k) = dist(gen);
-
+    {
+      auto in = make_host_view(field_in);
+      for(int i = 0; i < N; ++i)
+        for(int j = 0; j < M; ++j)
+          for(int k = 0; k < K; ++k)
+            in(i, j, k) = dist(gen);
```
Comment (stefanmoosbrugger) [inline examples/gridtools/simple/example-03-slice.cpp]: same as before.. you could use the lambda.. only if you want.. :)

File: test/utility/GridTools.h [hunk 4]
```diff
@@ -48,82 +46,56 @@ struct gridtools_storage_types {
...
-  using gpu_4d_storage_type = storage_traits_type::storage_type<T, gpu_4d_meta_data_type>;
+      storage_traits_type::special_storage_info_t<3, gridtools::selector<1, 1, 0>, halo_3d_type>;
+  using gpu_2d_meta_data_type = gridtools::host_storage_info<4, gpu_2d_layout_type, halo_3d_type,
+                                                             gridtools::alignment<gpu_alignment>>;
+  using cpu_3d_meta_data_type = storage_traits_type::storage_info_t<5, 3, halo_3d_type>;
+  using gpu_3d_meta_data_type = gridtools::host_storage_info<6, gpu_3d_layout_type, halo_3d_type,
+                                                             gridtools::alignment<gpu_alignment>>;
+  using cpu_4d_meta_data_type = storage_traits_type::storage_info_t<7, 4, halo_4d_type>;
+  using gpu_4d_meta_data_type = gridtools::host_storage_info<8, gpu_4d_layout_type, halo_4d_type,
+                                                             gridtools::alignment<gpu_alignment>>;
```
Comment (stefanmoosbrugger) [inline test/utility/GridTools.h]: why is `host_storage_info` used and the using decl is named `gpu_4d_meta_data_type`?

File: test/serialbox/core/frontend/gridtools/UnittestSerializer.cpp [hunk 5]
```diff
@@ -180,30 +180,30 @@ class GridToolsReadWriteTest : public SerializerUnittestBase {
...
-        gpu_4d_storage_output(gpu_4d_meta_data, "gpu_4d_storage_output", -1.0) {
+        cpu_2d_real_meta_data(dim1, dim2),                                             //
+        gpu_2d_real_meta_data(dim1, dim2),                                             //
+        cpu_2d_meta_data(dim1, dim2, 1),                                               //
+        gpu_2d_meta_data(dim1, dim2, 1),                                               //
+        cpu_3d_meta_data(dim1, dim2, dim3),                                            //
+        gpu_3d_meta_data(dim1, dim2, dim3),                                            //
+        cpu_4d_meta_data(dim1, dim2, dim3, dim4),                                      //
+        gpu_4d_meta_data(dim1, dim2, dim3, dim4),                                      //
+        cpu_2d_real_storage_input(cpu_2d_real_meta_data, "cpu_2d_real_storage_input"), //
```
Comment (stefanmoosbrugger) [inline test/serialbox/core/frontend/gridtools/UnittestSerializer.cpp]: no init needed anymore?

File: src/serialbox/core/frontend/gridtools/StorageViewHelper.h [hunk 6]
```diff
@@ -25,83 +25,34 @@ namespace gridtools {
...
-  int max_value = boost::mpl::deref<
-      typename boost::mpl::max_element<typename LayoutMap::layout_vector_t>::type>::type::value;
-
-  const int value = coord < (int)LayoutMap::length ? LayoutMap::layout_vector[coord] : -1;
-  return (max_value < 0) ? 0 : (value == max_value ? 1 : container[value + 1]);
+inline std::vector<int> unsigned_to_int(const std::vector<unsigned int>& in) {
+  std::vector<int> out;
+  for(const auto elem : in)
+    out.push_back(elem);
+  return out;
```
Comment (stefanmoosbrugger) [inline src/serialbox/core/frontend/gridtools/StorageViewHelper.h]:
```c++
std::vector<int> out(in.size());
std::copy(in.begin(), in.end(), out.begin());
return out;
```

File: test/serialbox/core/frontend/gridtools/UnittestSerializer.cpp (hunk 5 above)
Comment (havogt) [inline test/serialbox/core/frontend/gridtools/UnittestSerializer.cpp]: You mean I could put the init in the ctor?

**Comment by havogt** [comment]
launch jenkins

File: test/utility/GridTools.h [hunk 7]
```diff
@@ -48,82 +46,57 @@ struct gridtools_storage_types {
...
+  using cpu_4d_storage_type = storage_traits_type::data_store_t<T, cpu_4d_meta_data_type>;
+  using gpu_4d_storage_type = storage_traits_type::data_store_t<T, gpu_4d_meta_data_type>;
 
   template <typename Storage>
   static void init2DReal(Storage& storage, int dim1, int dim2) {
     T val_2d = 0.0;
     for(int j = 0; j < dim2; ++j)
       for(int i = 0; i < dim1; ++i, val_2d += 1.0) {
-        storage(i, j) = val_2d;
+        auto view = make_host_view(storage);
```
Comment (mbianco) [inline test/utility/GridTools.h]: making the view outside the loop?

File: test/serialbox/core/frontend/gridtools/UnittestSerializer.cpp (hunk 5 above)
Comment (mbianco) [reply test/serialbox/core/frontend/gridtools/UnittestSerializer.cpp]: Using lambdas to initialize would simplify the thing

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**